### ISSUE TITLE
Confusing signature for MessageFunction type

### ISSUE THREAD

**Issue by gamedevsam** [messageformat/messageformat#307]
According to the docs: http://messageformat.github.io/messageformat/api/core.messagefunction/

The MessageFunction type returns either a `string` or an `any[]`:

```ts
export declare type MessageFunction = (param?: object) => string | any[];
```

I was trying to figure out under what circumstance a message function will return an array instead of a string, but can't find any reference to that use case in the documentation.

Can someone clear this up for me?

**Comment by eemeli** [comment]
That's controlled by this option: http://messageformat.github.io/messageformat/api/core.messageformatoptions.returntype/

**Comment by gamedevsam** [comment]
Thanks, I've overwritten this type in my local project to always return string, it might make sense to provide extra types so users can use the right type for their use case, ex:

```ts
export declare type MessageFunction = (param?: object) => string | any[];
export declare type MessageFunctionString = (param?: object) => string;
export declare type MessageFunctionValues = (param?: object) => Array;
```

**Comment by eemeli** [comment]
I re-opened this so that it could get closed again by #312. 😇
